cloud/app.py

```diff
@@ -42,6 +44,26 @@ def collectBridgeMetrics(devices):
     return bridgedict
 
 
+def token_required(f):
+    @wraps(f)
+    def decorator(*args, **kwargs):
+
+        token = None
+        if 'x-access-tokens' in request.headers:
+            token = request.headers['x-access-tokens']
+
+        if not token:
+            return {'message': 'a valid token is missing'}
+
+        try:
+            data = jwt.decode(token, app.config['SECRET_KEY'], algorithms=["HS256"])
+            data['public_id'] in app.config['BRIDGE_TOKENS']
```

Comment:
This statement does not seem to have any effect? If it should only allow some bridges to connect, it does not really has any security effect. If an attacker steals the encrypted token of a bridge, they can do a replay attack and if they steal the secret key, they could create encrypted tokens for all bridge tokens.

Reply:
And it actually would have have a little effect when it would be used, it's possible to deprecate old tokens without changing the secret (as long as the person holding the token never saw the secret)

Reply:
Now it has more effect :+1: